exodus/analysis_query/models.py

```diff
@@ -46,11 +48,26 @@ def validate_handle(handle):
             _('Too much pending requests, please retry later'))
 
 
+def validate_file_extension(value):
+    import os
+    from django.core.exceptions import ValidationError
+    ext = os.path.splitext(value.name)[1]  # [0] returns path+filename
+    valid_extensions = ['.apk', ]
+    if not ext.lower() in valid_extensions:
+        raise ValidationError(_('Unsupported file extension.'))
```

Comment:
To be nitpicky, I would remove the `.` at the end of the error message :)

Reply:
Okay